.github/workflows/publish-docs-preview.yml

```diff
@@ -11,15 +11,40 @@ jobs:
     if: "${{ github.event.label.name == ':eyes: publish-docs-preview' }}"
     steps:
 
+      - name: Find CI workflow run for PR
+        id: find-workflow-run
+        uses: actions/github-script@v7
+        continue-on-error: true
+        with:
+          script: |
+            // Find the last successful workflow run for the current PR's head
+            const {owner, repo} = context.repo;
+            const runsResponse = await github.rest.actions.listWorkflowRuns({
+              owner,
+              repo,
+              workflow_id: 'ci.yml',
+              event: 'pull_request',
+              head_sha: context.sha,
+              status: 'success'
+            });
+            const runs = runsResponse.data.workflow_runs;
+            runs.sort((a, b) => b.created_at.getTime() - a.created_at.getTime());
+            const run = runs[0];

```

Comment:
I fear we'll get a nasty error if `runs` is empty. Which I guess could happen if someone (me) attached the label too quickly? Especially dangerous if the PR comes from an outside contributor for whom we haven't allow-listed the actions.

Reply:
> I fear we'll get a nasty error if `runs` is empty

No, it just returns `undefined`.

Reply:
blow me down